---
title: "Seurat: Quality control"
#CSS_ALL:
---

***
#DATA_TITLE:

#DATA_ALL1:

```{bash, results='hide'}
mkdir data
curl -o data/pbmc_1k_v2_filtered_feature_bc_matrix.h5 -O http://cf.10xgenomics.com/samples/cell-exp/3.0.0/pbmc_1k_v2/pbmc_1k_v2_filtered_feature_bc_matrix.h5
curl -o data/pbmc_1k_v3_filtered_feature_bc_matrix.h5 -O http://cf.10xgenomics.com/samples/cell-exp/3.0.0/pbmc_1k_v3/pbmc_1k_v3_filtered_feature_bc_matrix.h5
curl -o data/pbmc_1k_protein_v3_filtered_feature_bc_matrix.h5 -O http://cf.10xgenomics.com/samples/cell-exp/3.0.0/pbmc_1k_protein_v3/pbmc_1k_protein_v3_filtered_feature_bc_matrix.h5
```

#DATA_ALL2:

```{r, message='hide',warning='hide',results='hold'}
suppressMessages(require(DropletUtils))
suppressMessages(require(scater))
suppressMessages(require(Matrix))
suppressMessages(require(rhdf5))
```

#DATA_ALL3:

```{r,message='hide',warning='hide',results='hold'}
v3.1k <- Read10X_h5("data/pbmc_1k_v3_filtered_feature_bc_matrix.h5", use.names = T)
v2.1k <- Read10X_h5("data/pbmc_1k_v2_filtered_feature_bc_matrix.h5", use.names = T)
p3.1k <- Read10X_h5("data/pbmc_1k_protein_v3_filtered_feature_bc_matrix.h5", use.names = T)
p3.1k <- p3.1k$`Gene Expression`
```

***
#OBJ_TITLE:

#OBJ_ALL1:

```{r}
sce <- SingleCellExperiment( assays = list(counts = cbind(v3.1k,v2.1k,p3.1k)) )
sce
cpm(sce) <- calculateCPM(sce)
sce <- normalize(sce)
assayNames(example_sce)

sce@metadata$
```

#OBJ_ALL2:

```{r ,  results='hold'}


```


***
#QC_TITLE:

#QC_ALL1:

#QC_ALL1.1:

```{r, results='hold'}
sce <- calculateQCMetrics(sce)
```

#QC_ALL2:

```{r, results='hold'}


```

***
#QC_TITLE2:

#QC_ALL3:

```{r,message='hide',warning='hide', results='hold',results='hold',fig.height=5,fig.width=16}


```

#QC_ALL4:

```{r,message='hide',warning='hide', results='hold',results='hold',fig.height=5,fig.width=6}
plotHighestExprs(sce, exprs_values = "counts",n = 20)
```

***
#FILTERING_TITLE:

#FILTERING_TITLE1:

#FILTERING_ALL0:

```{r,message='hide',warning='hide', results='hold',results='hold',fig.height=5,fig.width=4}

sce@assays$
```

#FILTERING_ALL3:

```{r,message='hide',warning='hide', results='hold',results='hold',fig.height=5,fig.width=16}


```

#FILTERING_ALL01:

```{r,message='hide',warning='hide', results='hold',results='hold',fig.height=5,fig.width=12}
#Compute the relative expression of each gene per cell
rel_expression <- t( t(data.filt@assays$RNA@counts) / Matrix::colSums(data.filt@assays$RNA@counts)) * 100
most_expressed <- sort(Matrix::rowSums( rel_expression ),T)[20:1] / ncol(data.filt)

par(mfrow=c(1,2),mar=c(4,6,1,1))
boxplot( as.matrix(t(rel_expression[names(most_expressed),])),cex=.1, las=1, xlab="% total count per cell",col=scales::hue_pal()(20)[20:1],horizontal=TRUE)
```

#FILTERING_ALL02:

#FILTERING_TITLE2:

#FILTERING_ALL1:

```{r,message='hide',warning='hide', results='hold',results='hold',fig.height=5,fig.width=16}
selected_mito <- WhichCells(data.filt, expression = percent_mito < 0.25)
selected_ribo <- WhichCells(data.filt, expression = percent_ribo > 0.05)

# and subset the object to only keep those cells
data.filt <- subset(data.filt, cells = selected_mito)
data.filt <- subset(data.filt, cells = selected_ribo)
```

#FILTERING_ALL2:

#FILTERING_TITLE4:

#FILTERING_ALL5:

```{r,message='hide',warning='hide', results='hold',results='hold',fig.height=5,fig.width=16}
feats <- c("nFeature_RNA","nCount_RNA","percent_mito","percent_ribo")
cowplot::plot_grid(ncol = 1,
VlnPlot(data.filt, group.by= "orig.ident", features = feats, pt.size = 0.1,ncol = 4) + NoLegend())
```

#CELLCYCLE_TITLE:

#CELLCYCLE_ALL1:

```{r,message='hide',warning='hide', results='hold',results='hold',fig.height=4,fig.width=8}
data.filt <- NormalizeData(data.filt)

data.filt <- CellCycleScoring(object = data.filt,
                              g2m.features = cc.genes$g2m.genes,
                              s.features = cc.genes$s.genes)
```

#CELLCYCLE_ALL2:

```{r,message='hide',warning='hide', results='hold',results='hold',fig.height=5,fig.width=16}
VlnPlot(data.filt, features = c("S.Score","G2M.Score"), group.by= "orig.ident",ncol = 4, pt.size = .1)
```

#CELLCYCLE_ALL3:

#CELLCYCLE_ALL4:

```{r,message='hide',warning='hide', results='hold',results='hold',fig.height=5,fig.width=16}
saveRDS(data.filt,"data/3pbmc_qc.rds")
```






